File: testdata/jennies/builders/package-with-dashes/PythonBuilder/builders/builder_pkg.py
import typing
from ..cog import builder as cogbuilder
from ..models import with_dashes


class SomeNiceBuilder(cogbuilder.Builder[with_dashes.SomeStruct]):
    _internal: with_dashes.SomeStruct

    def __init__(self) -> None:
        self._internal = with_dashes.SomeStruct()

    def build(self) -> with_dashes.SomeStruct:
        """
        Builds the object.
        """
        return self._internal    
    
    def title(self, title: str) -> typing.Self:    
        self._internal.title = title
    
        return self
    

File: testdata/jennies/builders/package-with-dashes/RustBuilder/src/builders/builder_pkg.rs
use crate::cog;
use crate::types::with_dashes;

#[derive(Debug, Clone)]
pub struct SomeNiceBuilderBuilder {
    internal: with_dashes::SomeStruct,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl SomeNiceBuilderBuilder {
    pub fn new() -> Self {
        Self {
            internal: with_dashes::SomeStruct::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for SomeNiceBuilderBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl SomeNiceBuilderBuilder {
    pub fn title(mut self, title: String) -> Self {
        self.internal.title = title;

        self
    }
}

impl cog::Builder<with_dashes::SomeStruct> for SomeNiceBuilderBuilder {
    fn build(&self) -> Result<with_dashes::SomeStruct, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}
